tests/base_test_case.py

```diff
@@ -115,12 +115,16 @@ def assertRaisesRegexp(self, expected_exception, expected_regexp,
             self.fail("Expected exception " + str(expected_exception) +
                       " not raised" + msg)
 
+    @staticmethod
+    def assertLazyMessage(assert_function, actual, expected, msg_func):
+        try:
+            assert_function(actual, expected)
+        except AssertionError:
+            assert_function(actual, expected, msg=msg_func())
```

Comment:
Actually, that isn't really true, since the actual assert function won't show up. I was thinking there'd be some value in seeing what assertion you're making in the last frame, but you wouldn't, and it will be in the next frame up anyway.